### DISCUSSION TITLE
How should I style dropdown menus with dividers?

### DISCUSSION THREAD

**Discussion by seancolsen** [mathesar-foundation/mathesar#741] [Maintainer discussions]
- The ["Working with Columns" Figma Prototype](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?page-id=2750%3A17660&node-id=2816%3A18654&viewport=2205%2C368%2C0.5616371035575867&scaling=contain&starting-point-node-id=2816%3A18654) has a "Table Actions" menu as follows:

    ![image](https://user-images.githubusercontent.com/42411/138177534-946e73fb-482a-4a8c-b57a-abe7cc1fce7f.png)

- We currently have a similar menu for "New table" which behaves as follows:

    ![Peek 2021-10-20 17-52](https://user-images.githubusercontent.com/42411/138177731-b2a57bed-5951-4362-926d-660fd89bf0ea.gif)

- I like the divider between "Delete Table" and "Table Constraints" in the prototype, and I want to implement that. I also like the styling of the divider being partial width (narrower than the dropdown). **My predicament is: if we combine the hover and the divider I think it's going to look weird, having a full-width background and a partial-width divider.** It would help to see a mockup of a fully-styled dropdown menu that has a divider and has a menu item adjacent to the divider which is hovered. A description of the behavior would suffice as well.

@ghislaineguerin would you like to offer a recommendation?

**Comment by ghislaineguerin** [comment]
@seancolsen 
Let's keep the hover, add a margin around the menu items, and make all dividers partial-width. 
![MOXeB27fQo](https://user-images.githubusercontent.com/845767/138222899-0797ee3e-02ad-4672-bad4-d333ad6d1db4.gif)

**Reply by seancolsen** [reply]
Sounds good!
